clients/spark/core/src/main/hadoop3/scala/io/treeverse/clients/conditional/GCBackupAndRestore.scala

```diff
@@ -38,7 +46,13 @@ object GCBackupAndRestore {
     objectsRelativePathsDF
       .select("address")
       .as[String]
-      .flatMap(x => StorageUtils.concatKeysToStorageNamespace(Seq(x), storageNSForFS, storageType))
+      .flatMap(x =>
+        if (storageType == StorageUtils.StorageTypeS3) {
```

Comment:
What's wrong with the storage utils? The lack of default value?
Can add it as an option to the storage utils.